graphql_client_codegen/src/lib.rs

```diff
@@ -55,8 +55,10 @@ lazy_static! {
         CacheMap::default();
 }
 
-pub struct GraphQLClientDeriveOptions<'a> {
-    pub input: &'a syn::DeriveInput,
+pub struct GraphQLClientDeriveOptions {
+    pub selected_operation: String,
+    pub additional_derives: Option<String>,
+    pub deprecation_strategy: Option<deprecation::DeprecationStrategy>,
```

Comment:
Since there is default (`Warn`) I think `deprecation_strategy` does not need to be an option, what do you think?

Reply:
I actually think there is a "bug". The default on the struct is "allow" but in usage is "warn". I'll make them the same in a PR in the morning.

Reply:
> I think deprecation_strategy does not need to be an option, what do you think?

Now, we determine default value in the method, [generate_module_token_stream](https://github.com/tomhoule/graphql-client/pull/122/files/0b00534b6e037cb1dc8ebeed38dfb886745e5e98#diff-4cc81fc19b88d044ccd920473bc5492fR81).
So I let this field option.